tle/util/cache_system2.py

```diff
@@ -239,6 +239,13 @@ def keep(problem):
         self.problem_by_name = problem_by_name
         self.problems_last_cache = time.time()
 
+        for problem in self.problems:
+            problem_contest: cf.Contest = self.cache_master.contest_cache.contest_by_id.get(problem.contestId)
+
+            for i in range(1, 5):
```

Comment:
The code in both functions is doing the exact same, why is it so different :-). 
I also look for a more pythonic way to do it (for loops are for C++ devs :D)
Something like:
`divisions = [div_tag for div_tag in _DIV_TAGS if contest.matches(div_tag)]
 problem.tags.append(divisions)
`
And _DIV_TAGS can be declared globally for both functions.
` _DIV_TAGS = ['div1', 'div2', 'div3', 'div4']`